### ISSUE TITLE
`rename` duplicates a column

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#13766]
### Describe the bug

```Nushell
~> 1..=10
	| prepend 0 | enumerate | skip 1
	| rename you
───┬─────┬──────
 # │ you │ item 
───┼─────┼──────
 0 │   1 │    1 
 1 │   2 │    2 
 2 │   3 │    3 
 3 │   4 │    4 
 4 │   5 │    5 
 5 │   6 │    6 
 6 │   7 │    7 
 7 │   8 │    8 
 8 │   9 │    9 
 9 │  10 │   10 
───┴─────┴──────
```

### How to reproduce

-

### Expected behavior

-

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.97.1                                                       |
| major              | 0                                                            |
| minor              | 97                                                           |
| patch              | 1                                                            |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06) (Alpine Linux 1.80.1-r0) |
| cargo_version      | cargo 1.80.1 (376290515 2024-07-16)                          |
| build_time         | 2024-08-29 21:40:07 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, system-clipboard, trash                     |
| installed_plugins  | endecode 0.2.0, template 0.1.0                               |

**Comment by fdncred** [comment]
I think this is because `enumerate` returns and index and an item. So, you have to do this.
```nushell
❯ 1..=10 | prepend 0 | enumerate | rename index you
╭─#──┬─you─╮
│ 0  │   0 │
│ 1  │   1 │
│ 2  │   2 │
│ 3  │   3 │
│ 4  │   4 │
│ 5  │   5 │
│ 6  │   6 │
│ 7  │   7 │
│ 8  │   8 │
│ 9  │   9 │
│ 10 │  10 │
╰─#──┴─you─╯
```

**Comment by sholderbach** [comment]
Yeah not a bug as described but rather curious behavior based on our choices, `rename` when given less than the available columns will start from the beginning. And in this case its `index` from enumerate before `item`.

If you just output the result from `enumerate` with a default config the `index` column will be treated by `table` the same as its internal numbering.

For me the quesiton becomes is the `table` `index` column detection really worth it? i.e. detecting `index` keys incoming and treating them the same as the default numbering added by `table` with the respective index setting. Ripe for confusion in my view.

**Comment by KAAtheWiseGit** [comment]
Got it!  And I think I have an idea about this
